wire/decoder.go

```diff
@@ -0,0 +1,367 @@
+package wire
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"math"
+	"net"
+	"time"
+)
+
+//integer sizes
+const (
+	ByteSize    = 1
+	ShortSize   = 2
+	IntegerSize = 4
+	LongSize    = 8
+)
+
+// ConnInfo contains information about the database connection. This is returned
+// from the database on a successful login.
+type ConnInfo struct {
+
+	// Host is the host ID of the volt node
+	Host int32
+
+	// Connection unique id for the connection in the current volt node
+	Connection int64
+
+	// Time  in which the cluster was started
+	ClusterStart time.Time
+
+	// IPV4 address of the leader node
+	LeaderAddr struct {
+		Value int32
+		IP    net.IP
+	}
+
+	// Representation of the build of the connected node
+	Build string
+}
+
+// Decoder is voltdb wire protocol decoder
+type Decoder struct {
+	r io.Reader
```

Comment:
`io.Reader` makes more sense to me we are not tying to any particular source as it should be.

The premise is, `Decoder`  reades `[]byte` and interpret them to their respective native data types without caring where the bytes come from.